security/pkg/caclient/keycertbundlerotator_test.go

```diff
@@ -20,27 +20,40 @@ import (
 	"testing"
 	"time"
 
-	caclientmock "istio.io/istio/security/pkg/caclient/mock"
 	pkiutil "istio.io/istio/security/pkg/pki/util"
 	pkimock "istio.io/istio/security/pkg/pki/util/mock"
 	"istio.io/istio/security/pkg/util"
 	utilmock "istio.io/istio/security/pkg/util/mock"
 )
 
+type fakeKeyCertRetriever struct {
+	NewCert    []byte
+	CertChain  []byte
+	PrivateKey []byte
+	Err        error
+}
+
+func (r *fakeKeyCertRetriever) Retrieve() (newCert, certChain, privateKey []byte, err error) {
```

Comment:
Is there any benefit of moving the mocks to here?

Reply:
This test is the only one needs the fake, so I move it here & make it private. It reduces the public types we expose.

Reply:
OK. Thanks for clarification.